swap/actions.go

```diff
@@ -368,6 +368,16 @@ func (c *CreateSwapOutFromRequestAction) Execute(services *SwapServices, swap *S
 		return swap.HandleError(err)
 	}
 
+	// Check if onchain balance is sufficient for swap + fees + some safety net
+	walletBalance, err := wallet.GetOnchainBalance()
+	if err != nil {
+		return swap.HandleError(err)
+	}
+
+	if walletBalance < swap.GetAmount()+openingFee+20000 {
```

Comment:
Fine for now but maybe add TODO to comment that this safety buffer be calculated to be more precise. Otherwise sometimes people will wonder why a swap is failing when they think they have enough.